includes/class-newspack-newsletters-contacts.php

```diff
@@ -17,24 +17,19 @@ class Newspack_Newsletters_Contacts {
 	/**
 	 * Upserts a contact to lists.
 	 *
-	 * A contact can be added asynchronously, which means the request will return
-	 * immediately and the contact will be added in the background. In this case
-	 * the response will be `true` and the caller must handle it optimistically.
-	 * NEWSPACK_NEWSLETTERS_ASYNC_SUBSCRIPTION_ENABLED must be defined as true for this feature to be available.
-	 *
-	 * @param array          $contact {
+	 * @param array          $contact          {
 	 *          Contact information.
 	 *
 	 *    @type string   $email    Contact email address.
 	 *    @type string   $name     Contact name. Optional.
 	 *    @type string[] $metadata Contact additional metadata. Optional.
 	 * }
-	 * @param string[]|false $lists   Array of list IDs to subscribe the contact to. If empty or false, contact will be created but not subscribed to any lists.
-	 * @param bool           $async   Whether to add the contact asynchronously. Default is false.
+	 * @param string[]|false $lists            Array of list IDs to subscribe the contact to. If empty or false, contact will be created but not subscribed to any lists.
+	 * @param array|null     $existing_contact Optional existing contact data.
 	 *
 	 * @return array|WP_Error|true Contact data if it was added, or error otherwise. True if async.
 	 */
-	public static function upsert( $contact, $lists = false, $async = false ) {
+	public static function upsert( $contact, $lists = false, $existing_contact = null ) {
```

Comment:
The async strategy was removed from this method to be handled by the new `subscribe_contact` method. The async strategy is meant to improve UX exclusive to newsletter subscriptions and should not be available for every upsert.

The `$existing_contact` was added as a parameter in case the caller already fetched the data, which is the case for the new `subscribe_contact` method.